### PR TITLE
depend on latest nipype master

### PR DESCRIPTION
**Pull request by shoshber**

### CODE REVIEW

**Comment by oesteban** [comment]
can you rebase, so I can merge this?

**Comment by shoshber** [comment]
I pulled from poldracklab's master--that's what you mean, right?

**Comment by chrisfilo** [comment]
[Merging master is slightly different from rebasing](https://www.atlassian.com/git/tutorials/merging-vs-rebasing/workflow-walkthrough), but will solve the problem as well.

File: fmriprep/run_workflow.py [hunk 1]
```diff
@@ -28,7 +30,7 @@ def main():
     from nipype.pipeline import engine as pe
     from fmriprep import __version__
     from fmriprep.workflows import fmriprep_single
-
+    from fmriprep.utils.misc import collect_bids_data
```
Comment (oesteban) [inline fmriprep/run_workflow.py]: The collect_bids_data is not used here anymore, please remove.

File: fmriprep/run_workflow.py [hunk 2]
```diff
@@ -80,8 +76,12 @@ def main():
         'work_dir': op.abspath(opts.work_dir)
     }
 
+    # set up logger
+    logger = logging.getLogger('cli')
+
     if opts.debug:
         settings['ants_t1-mni_settings'] = 't1-mni_registration_test'
+        logger.setLevel(logging.DEBUG)
```
Comment (oesteban) [inline fmriprep/run_workflow.py]: :+1: this is a great idea :)

File: fmriprep/utils/misc.py [hunk 3]
```diff
@@ -63,38 +61,44 @@ def collect_bids_data(dataset, subject, spec=None, session=None, run=None):
...
+
+        session_data = copy.deepcopy(INPUTS_SPEC)
+        fieldmap_files = [x.filename for x in layout.get(**queries['fmap'])]
+        session_data['fmap'] = fieldmap_files
+        t1_files = [x.filename for x in layout.get(**queries['t1w'])]
+        session_data['t1w'] = t1_files
+        sbref_files = [x.filename for x in layout.get(**queries['sbref'])]
+        session_data['sbref'] = sbref_files
+
+        loop_on = ['session', 'run', 'acquisition', 'task']
```
Comment (oesteban) [inline fmriprep/utils/misc.py]: I think @rwblair was serializing sessions as well with this loop. I think this file deserves a different PR where he can check it out and you can go through it together.

File: test/utils/test_misc.py [hunk 4]
```diff
@@ -11,11 +12,12 @@ class TestCollectBids(unittest.TestCase):
...
-            all_subjects = misc.collect_bids_data(c.DATASET)
+            subject_data = misc.collect_bids_data(c.DATASET, self.subject_id,
+                                                  c.BIDS_SPEC)
             self.imaging_data = {
-                self.subject_id: all_subjects[self.subject_id]
+                self.subject_id: subject_data
             }
-        except Exception as e:
+        except IOError as e:
             url = "http://googledrive.com/host/0BxI12kyv2olZbl9GN3BIOVVoelE"
```
Comment (oesteban) [inline test/utils/test_misc.py]: Please update the link with the new one (see #104)

**Comment by oesteban** [comment]
This PR is great. My only concern is ```fmriprep/utils/misc.py```. I think @rwblair had already given support for multisession datasets in it. So let's split it, hold this file off the PR and submit it in another different PR that Ross will review when he is back with you. Sounds good?

File: test/utils/test_misc.py (hunk 4 above)
Comment (shoshber) [inline test/utils/test_misc.py]: I will do this in another pr

File: fmriprep/utils/misc.py (hunk 3 above)
Comment (shoshber) [inline fmriprep/utils/misc.py]: I had to do this to get the tests to work. Can we commit this for now?

File: fmriprep/run_workflow.py (hunk 1 above)
Comment (shoshber) [inline fmriprep/run_workflow.py]: done

**Comment by oesteban** [comment]
Ok, I see.

The problem here is that tests are anchored to the old downsampled dataset, right? (https://github.com/shoshber/preprocessing-workflow/blob/nipypedep87/test/utils/test_misc.py#L15). Let's think a bit about this.

We basically need to test two things:

  1. We need to make sure that fmriprep correctly builds up the list of subjects x sessions x image-sets to be passed on the workflow. This means testing the ```collect_bids_data``` function. This is why #103 exists.
  2. Smoke tests on the pipeline itself (#107) - BTW I'm going to edit this issue to make it more representative of the idea.

Basically, you need to split the current output list from ```collect_bids_data``` into sessions, so that it works with the old dataset, which was intended for the smoke tests in 2).

I would propose to forget about fixing circleci right now (we'll keep the annoying reds for a while), and merge this PR which intention was only _depend on latest nipype master_.

Next steps would be: #103 (there is no need for real/downsampled data to make this real) and then #107 (this needs the dataset that @suyashdb arranged).

**Comment by shoshber** [comment]
Alright, that makes sense. I think I can cherry-pick just the relevant commits to leave unit tests/collect_bids_data untouched. Didn't see #103 until now.

**Comment by shoshber** [comment]
closing this, see new pr
